programs/marinade-finance/src/instructions/delayed_unstake/order_unstake.rs

```diff
@@ -37,46 +37,29 @@ pub struct OrderUnstake<'info> {
 }
 
 impl<'info> OrderUnstake<'info> {
-    fn check_burn_msol_from(&self, msol_amount: u64) -> Result<()> {
-        if self
-            .burn_msol_from
-            .delegate
-            .contains(self.burn_msol_authority.key)
-        {
-            // if delegated, check delegated amount
-            // delegated_amount & delegate must be set on the user's msol account before calling OrderUnstake
-            require_lte!(
-                msol_amount,
-                self.burn_msol_from.delegated_amount,
-                MarinadeError::NotEnoughUserFunds
-            );
-        } else if self.burn_msol_authority.key() == self.burn_msol_from.owner {
-            require_lte!(
-                msol_amount,
-                self.burn_msol_from.amount,
-                MarinadeError::NotEnoughUserFunds
-            );
-        } else {
-            return err!(MarinadeError::WrongTokenOwnerOrDelegate).map_err(|e| {
-                e.with_account_name("burn_msol_from")
-                    .with_pubkeys((self.burn_msol_from.owner, self.burn_msol_authority.key()))
-            });
-        }
-        Ok(())
-    }
 
     // fn order_unstake() // create delayed-unstake Ticket-account
     pub fn process(&mut self, msol_amount: u64) -> Result<()> {
         require!(!self.state.paused, MarinadeError::ProgramIsPaused);
 
-        self.check_burn_msol_from(msol_amount)?;
+        check_token_source_account(
+            &self.burn_msol_from,
+            self.burn_msol_authority.key,
+            msol_amount,
+        )?;
         let ticket_beneficiary = self.burn_msol_from.owner;
         let user_msol_balance = self.burn_msol_from.amount;
 
         // save msol price source
         let total_virtual_staked_lamports = self.state.total_virtual_staked_lamports();
         let msol_supply = self.state.msol_supply;
-        let lamports_amount = self.state.calc_lamports_from_msol_amount(msol_amount)?;
+
+        let sol_value = self.state.calc_lamports_from_msol_amount(msol_amount)?;
+        // apply delay_unstake_fee to avoid economical attacks
+        // delay_unstake_fee must be >= one epoch staking rewards
+        let delay_unstake_fee_lamports = self.state.delayed_unstake_fee.apply(sol_value);
+        // the fee value will be burned but not delivered, thus increasing mSOL value slightly for all mSOL holders
+        let lamports_amount = sol_value - delay_unstake_fee_lamports;
```

Comment:
Names sol_value and lamport_amount are not descriptive. You need to select one of 2 ways: 1) use the same name shadowing the old variable 2) make descriptive names like sol_amount_after_fee_applied or find better words somehow

Reply:
changed them, propose others, no prob ussing your suggestions